components/brave_rewards/resources/tip/lib/host.ts

```diff
@@ -0,0 +1,225 @@
+        return
+      }
+      for (const tip of tips) {
+        if (tip.publisherKey === dialogArgs.publisherKey) {
+          stateManager.update({
+            currentMonthlyTip: Number(tip.amount) || 0
+          })
+          break
+        }
+      }
+    },
+
+    reconcileStampUpdated (stamp: number) {
+      stateManager.update({ nextReconcileDate: new Date(stamp * 1000) })
+    },
+
+    balanceUpdated (result: { status: number, balance: BalanceInfo }) {
+      if (result.status === 0) {
+        stateManager.update({
+          balanceInfo: result.balance
+        })
+      } else {
+        stateManager.update({
+          hostError: {
+            type: 'ERROR_FETCHING_BALANCE',
+            code: result.status
+          }
+        })
+      }
+    },
+
+    onlyAnonWalletUpdated (onlyAnonWallet: boolean) {
+      stateManager.update({ onlyAnonWallet })
+    },
+
+    recurringTipRemoved (success: boolean) {
+      chrome.send('getRecurringTips')
+    },
+
+    recurringTipSaved (success: boolean) {
+      chrome.send('getRecurringTips')
+    },
+
+    reconcileCompleted (data: { type: number, result: number }) {
+      if (data.result === 0) {
+        chrome.send('fetchBalance')
+      }
+    }
+
+  })
+
+  self.i18nTemplate.process(document, self.loadTimeData)
+
+  chrome.send('dialogReady')
+
+  return {
+
+    get state () {
+      return stateManager.getState()
+    },
+
+    getString (key: StringKey) {
+      return self.loadTimeData.getString(key)
+    },
+
+    getDialogArgs () {
+      return dialogArgs
+    },
+
+    closeDialog () {
+      chrome.send('dialogClose')
+    },
+
+    processTip (amount: number, kind: TipKind) {
+      if (!dialogArgs.publisherKey) {
+        stateManager.update({
+          hostError: { type: 'INVALID_PUBLISHER_KEY' }
+        })
+        return
+      }
+
+      if (amount < 0 || kind === 'one-time' && amount === 0) {
+        stateManager.update({
+          hostError: { type: 'INVALID_TIP_AMOUNT' }
+        })
+        return
+      }
+
+      chrome.send('onTip', [
+        dialogArgs.publisherKey,
+        amount,
+        kind === 'monthly' ? true : false
+      ])
+
+      stateManager.update({
+        tipProcessed: true,
+        tipAmount: amount
+      })
+    },
+
+    shareTip (target: ShareTarget) {
+      const { publisherInfo } = stateManager.getState()
+      if (!publisherInfo) {
+        stateManager.update({
+          hostError: { type: 'PUBLISHER_INFO_UNAVAILABLE' }
+        })
+        return
+      }
+
+      let name = publisherInfo.name
+      let tweetId = ''
+
+      if (dialogArgs.mediaMetaData.mediaType === 'twitter') {
+        name = '@' + dialogArgs.mediaMetaData.screenName
+        tweetId = dialogArgs.mediaMetaData.tweetId
+      } else if (publisherInfo.provider === 'twitter' && dialogArgs.url) {
+        name = '@' + dialogArgs.url.replace(/^.*\//, '')
+      }
+
+      chrome.send('tweetTip', [name, tweetId])
+      chrome.send('dialogClose')
```

Comment:
Not sure if it's a matter for this specific PR, but it doesn't seem like we would need to send two separate commands for this.

Reply:
I think `dialogClose` is built-in to webUI and is also used for the close "x" button. I suppose I could figure out how to close the dialog from the "tweetTip" handler, but this seems just as good to me.